src/main/java/gregtech/api/capability/data/query/DataQueryObject.java

```diff
@@ -0,0 +1,29 @@
+package gregtech.api.capability.data.query;
+
+public abstract class DataQueryObject {
+
+    private static int ID = 0;
+
+    private final int id;
+
+    private boolean shouldTriggerWalker = false;
+
+    public DataQueryObject() {
+        this.id = ID++;
+    }
+
+    public void setShouldTriggerWalker(boolean shouldTriggerWalker) {
+        this.shouldTriggerWalker = shouldTriggerWalker;
+    }
+
+    public boolean shouldTriggerWalker() {
+        return shouldTriggerWalker;
+    }
+
+    public abstract DataQueryFormat getFormat();
+
+    @Override
+    public int hashCode() {
```

Comment:
DataQueryObjects should never be equal to each other; the internally incrementing ID passed to hash is specifically to prevent map collisions as much as possible